src/SmartTransactionsController.ts

```diff
@@ -330,42 +395,60 @@ export default class SmartTransactionsController extends BaseController<
     });
   }
 
-  async updateSmartTransactions() {
+  async updateSmartTransactions({
+    networkClientId,
+  }: {
+    networkClientId?: NetworkClientId;
+  } = {}): Promise<void> {
     const { smartTransactions } = this.state.smartTransactionsState;
-    const { chainId } = this.config;
-
-    const currentSmartTransactions = smartTransactions?.[chainId];
+    const chainId = this.getChainId({ networkClientId });
+    const smartTransactionsForChainId = smartTransactions?.[chainId];
 
-    const transactionsToUpdate: string[] = currentSmartTransactions
+    const transactionsToUpdate: string[] = smartTransactionsForChainId
       .filter(isSmartTransactionPending)
       .map((smartTransaction) => smartTransaction.uuid);
 
     if (transactionsToUpdate.length > 0) {
-      this.fetchSmartTransactionsStatus(transactionsToUpdate);
+      this.fetchSmartTransactionsStatus(transactionsToUpdate, {
+        networkClientId,
+      });
     }
   }
 
-  async confirmSmartTransaction(smartTransaction: SmartTransaction) {
+  // TODO make this a private method?
```

Comment:
🤔

Reply:
I think we should double-check that this isn't used anywhere. Regardless it seems okay to leave for now. Maybe we remove this TODO for now (unless we are getting to it soon) and ticket this instead? Your call on how you want to keep track of it though

Reply:
Update: I just double-checked this myself. Let's make this private!

Reply:
done here: https://github.com/MetaMask/smart-transactions-controller/pull/210/commits/99fc2cc69f1b6a520d23b3eb77f092c883305a78